pkg/sif/write.go

```diff
@@ -94,8 +100,13 @@ func blobFromIndex(ii v1.ImageIndex, digest v1.Hash) (io.ReadCloser, error) {
 	return nil, errUnableToReadBlob
 }
 
-// writeIndexToFileImage writes an index and all of its child indexes, manifests and blobs to f.
-func (f *fileImage) writeIndexToFileImage(ii v1.ImageIndex, rootIndex bool) error {
+// WriteIndex writes an index and all of its child indexes, manifests and blobs
+// to f. This function does not update the RootIndex.
+func (f *OCIFileImage) WriteIndex(ii v1.ImageIndex) error {
```

Comment:
Wondering if we might want to hold off exposing this... at some point we likely want a `name.Reference` to be provided, which will be associated with the `ImageIndex`?

Reply:
I was keeping to the pattern of layout.Path here, where the WriteXXXX are 'dumb' functions that just write into the blobs folder, and don't touch the RootIndex. Since associating a tag is an operation on the RootIndex, to add a descriptor with the correct annotation, the current exposure is consistent.

The AppendXXXX functions take options, so AppendIndex in https://github.com/sylabs/oci-tools/issues/83 would be the place that you'd specify a name.Reference that would become an annotation in the RootIndex.

If you'd prefer to deviate from the Write vs Append split as it is in ggcr layout.Path, I'm fine with that, though.

Reply:
Actually - perhaps holding of exposing this does make sense, as you say.... we talked a lot in the Update work  about avoiding 'holes' in the SIF where possible when the RootIndex has to be updated. Update was written and used in a way that makes sure the RootIndex is always at the end.

Will make this private for now.

Reply:
Made private in [f6e9323](https://github.com/sylabs/oci-tools/pull/86/commits/f6e9323bc8d643872f0dbff562c7f47eee6a0f23)